### PR TITLE
Feature Spotlight: Advanced Plugins

### PR DESCRIPTION
**Pull request by misteroneill**

### CODE REVIEW

File: source/_drafts/Feature-Spotlight-Advanced-Plugins.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
+---
+title: 'Feature Spotlight: Advanced Plugins'
+author:
+  name: "Pat O'Neill"
+  github: 'misteroneill'
+tags:
+  - plugins
+---
+
+> **Note:** Advanced plugins are being introduced in Video.js 6.0 and are only supported from that version forward.
```
Comment (gkatsev) [inline source/_drafts/Feature-Spotlight-Advanced-Plugins.md]: this should be a [pullquote](https://hexo.io/docs/tag-plugins.html#Pull-Quote)

File: source/_drafts/Feature-Spotlight-Advanced-Plugins.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
+---
+title: 'Feature Spotlight: Advanced Plugins'
+author:
+  name: "Pat O'Neill"
+  github: 'misteroneill'
+tags:
+  - plugins
```
Comment (gkatsev) [inline source/_drafts/Feature-Spotlight-Advanced-Plugins.md:7]: add `video.js 6.0` tag.

File: source/_drafts/Feature-Spotlight-Advanced-Plugins.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+One of Video.js' strengths is its rich ecosystem of plugins; so, in the last few months, we wanted to focus our efforts on improving the plugin author experience.
+
+While projects like [the plugin generator][generator] and [videojs-spellbook][spellbook] make becoming a plugin author easier than ever, the Video.js team thought it was important to provide a foundational API and set of conventions on which the future of Video.js plugins could be built.
+
+Our solution is _advanced plugins_.
+
+### Advanced Plugins are Component-Like
+
+One of the design goals for advanced plugins was to provide an API that was reminiscent of the existing components system. We achieved this in a number of ways.
```
Comment (gkatsev) [inline source/_drafts/Feature-Spotlight-Advanced-Plugins.md:33]: should we mention react for buzz word bingo (and seo)?

File: source/_drafts/Feature-Spotlight-Advanced-Plugins.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+The `helloWorld` method will return this plugin object until it is disposed - after which it will create a new plugin instance again.
+
+### Events
+
+Similar to components, advanced plugins can listen to and trigger events via the `on`, `one`, `off`, and `trigger` methods.
+
+This provides a loosely coupled communication channel for plugins and other objects (components, players, etc) to manage their own state and respond to changes in the state of one another.
+
+#### Plugin Event Hash
```
Comment (gkatsev) [inline source/_drafts/Feature-Spotlight-Advanced-Plugins.md]: Is this something we used in the videojs docs? The term seems new and unfamiliar. Could be substitute `hash` with something simpler that reaches a wider audience?

File: source/_drafts/Feature-Spotlight-Advanced-Plugins.md (hunk 3 above)
Comment (gkatsev) [reply source/_drafts/Feature-Spotlight-Advanced-Plugins.md:33]: I guess that's below :D

File: source/_drafts/Feature-Spotlight-Advanced-Plugins.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+  }
+});
+
+// Logs: 'The helloWorld color changed from "red" to "blue"!'
+instance.setState({color: 'blue'}); 
+```
+
+## Player Plugin Awareness
+
+Finally, we couldn't add new plugin infrastructure without working on one of the more pernicious problems of managing complex combinations of plugins: the player can't report which plugins it has initialized - or not. To this end, the player has two new methods: `hasPlugin` and `usingPlugin`. These methods work for _both_ types of plugins.
```
Comment (gkatsev) [inline source/_drafts/Feature-Spotlight-Advanced-Plugins.md:174]: "pernicious" -- fancy :P

**Comment by gkatsev** [comment]
Some minor comments, looks great!

File: source/_drafts/Feature-Spotlight-Advanced-Plugins.md (hunk 4 above)
Comment (misteroneill) [reply source/_drafts/Feature-Spotlight-Advanced-Plugins.md]: Sure. That is just what it's called in the code. I'll see if there's other words used in other places.

Comment (misteroneill) [reply source/_drafts/Feature-Spotlight-Advanced-Plugins.md]: Renaming this to "Additional Event Data".

File: source/_drafts/Feature-Spotlight-Advanced-Plugins.md (hunk 2 above)
Comment (misteroneill) [reply source/_drafts/Feature-Spotlight-Advanced-Plugins.md:7]: Done.

**Review by gkatsev — APPROVED** [review]

File: source/_drafts/Feature-Spotlight-Advanced-Plugins.md [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+title: 'Feature Spotlight: Advanced Plugins'
+author:
+  name: "Pat O'Neill"
+  github: 'misteroneill'
+tags:
+  - plugins
+  - video.js 6.0
+---
+
+{% pullquote %}
```
Comment (mjneil) [inline source/_drafts/Feature-Spotlight-Advanced-Plugins.md]: this isn't rendering properly

**Review by gkatsev — APPROVED** [review]

**Pull request merged**